java/algorithm/src/test/java/org/apache/arrow/algorithm/dictionary/TestSearchTreeBasedDictionaryBuilder.java

```diff
@@ -0,0 +1,179 @@
+      SearchTreeBasedDictionaryBuilder<VarCharVector> dictionaryBuilder =
+              new SearchTreeBasedDictionaryBuilder<>(dictionary, comparator, encodeNull);
+
+      int result = dictionaryBuilder.addValues(vec);
+
+      if (encodeNull) {
+        assertEquals(7, result);
+        assertEquals(7, dictionary.getValueCount());
+
+        dictionaryBuilder.populateSortedDictionary(sortedDictionary);
+
+        assertTrue(sortedDictionary.isNull(0));
+        assertEquals("12", new String(sortedDictionary.get(1)));
+        assertEquals("abc", new String(sortedDictionary.get(2)));
+        assertEquals("dictionary", new String(sortedDictionary.get(3)));
+        assertEquals("good", new String(sortedDictionary.get(4)));
+        assertEquals("hello", new String(sortedDictionary.get(5)));
+        assertEquals("world", new String(sortedDictionary.get(6)));
+      } else {
+        assertEquals(6, result);
+        assertEquals(6, dictionary.getValueCount());
+
+        dictionaryBuilder.populateSortedDictionary(sortedDictionary);
+
+        assertEquals("12", new String(sortedDictionary.get(0)));
+        assertEquals("abc", new String(sortedDictionary.get(1)));
+        assertEquals("dictionary", new String(sortedDictionary.get(2)));
+        assertEquals("good", new String(sortedDictionary.get(3)));
+        assertEquals("hello", new String(sortedDictionary.get(4)));
+        assertEquals("world", new String(sortedDictionary.get(5)));
+      }
+    }
+  }
+
+  @Test
+  public void testBuildFixedWidthDictionary() {
+    try (IntVector vec = new IntVector("", allocator);
+         IntVector dictionary = new IntVector("", allocator);
+         IntVector sortedDictionary = new IntVector("", allocator)) {
+      vec.allocateNew(10);
+      vec.setValueCount(10);
+
+      dictionary.allocateNew();
+      sortedDictionary.allocateNew();
+
+      // fill data
+      vec.set(0, 4);
+      vec.set(1, 8);
+      vec.set(2, 32);
+      vec.set(3, 8);
+      vec.set(4, 16);
+      vec.set(5, 32);
+      vec.setNull(6);
+      vec.set(7, 4);
+      vec.set(8, 4);
+      vec.setNull(9);
+
+      VectorValueComparator<IntVector> comparator = DefaultVectorComparators.createDefaultComparator(vec);
+      SearchTreeBasedDictionaryBuilder<IntVector> dictionaryBuilder =
+              new SearchTreeBasedDictionaryBuilder<>(dictionary, comparator, encodeNull);
+
+      int result = dictionaryBuilder.addValues(vec);
+
+      if (encodeNull) {
+        assertEquals(5, result);
+        assertEquals(5, dictionary.getValueCount());
+
+        dictionaryBuilder.populateSortedDictionary(sortedDictionary);
+
+        assertTrue(sortedDictionary.isNull(0));
+        assertEquals(4, sortedDictionary.get(1));
+        assertEquals(8, sortedDictionary.get(2));
+        assertEquals(16, sortedDictionary.get(3));
+        assertEquals(32, sortedDictionary.get(4));
+      } else {
```

Comment:
it doesn't seem like it makes much sense to parametererize these tests if the verifications are all different.  Either expected values should be parameters as well or the test should be split up.  I think they might clearer if the tests are split up.

Reply:
Sure. Tests are split up.